### DISCUSSION TITLE
feat(router): Support route groups

### DISCUSSION THREAD

**Discussion by nazo6** [wakujs/waku#561] [Ideas]
I think it would be useful to have route groups like Next.js that can separate layouts without affecting the URL path.

Next.js docs:
https://nextjs.org/docs/app/building-your-application/routing/route-groups

**Comment by dai-shi** [comment]
With `createPages` config router, I think it's possible. Please give it a try.

I'm not sure how we do with fs router, but we probably don't add such complexity. cc: @sandren 
(maybe supporting dynamic layout would be a solution.)

**Reply by nazo6** [reply]
Thanks for the response. However, I am not quite sure how to do this using createPages.

My understanding is that layout always applies to lower level components, so I thought it would not be possible to create a layout that would be visible under `/` or `/foo` but not `/bar`.

In next.js, this can be achieved by creating `(global-layout)/foo/page.tsx` and `(no-global-layout)/bar/page.tsx`.

**Reply by dai-shi** [reply]
You want to just `createPage`.

```js
  createPage({
    render: 'static',
    path: '/foo',
    component: () => <GlobalLayout><FooPage /></GlobalLayout>,
  });

  createPage({
    render: 'static',
    path: '/bar',
    component: () => <GlobalLayout><BarPage /></GlobalLayout>,
  });
```

**Reply by nazo6** [reply]
Ah, I can't believe I didn't think of this.
Thank you very much!
